### PR TITLE
JSUI-2606 Reset filters when the user enters a new query

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/JSUI-2606

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/Omnibox/Omnibox.ts [hunk 1]
```diff
@@ -260,7 +262,12 @@ export class Omnibox extends Component {
     querySuggestCharacterThreshold: ComponentOptions.buildNumberOption({
       defaultValue: 0,
       min: 0
-    })
+    }),
+
+    /**
+     * Specify wether all active query filters should be cleared everytime the user inputs a new query.
```
Comment (fbeaudoincoveo) [inline src/ui/Omnibox/Omnibox.ts]: Setting this to `true` would not clear the filter expression enforced by the currently selected tab though, correct?

Essentially, it clears the breadcrumbs?

If so, I would say:

> Whether to clear all active query filters when the end user submits a new query from the search box.
>
> **Note:** This does not include the filter expression enforced by the currently selected tab, if any.
>
> **Default:** `false`

File: src/ui/Omnibox/Omnibox.ts [hunk 2]
```diff
@@ -692,6 +704,30 @@ export class Omnibox extends Component {
...
+  }
+
+  private clearFiltersIfNewQuery({ origin, searchAsYouType }: INewQueryEventArgs) {
+    if (this.queryController.firstQuery) {
+      return;
+    }
+
+    // Prevent queries triggered by unrelated components to clear the the filters
+    // e.g., a facet selection
+    const validOrigins = [Omnibox.ID, 'SearchButton'];
```
Comment (olamothe) [inline src/ui/Omnibox/Omnibox.ts]: Should also include `Querybox` here.

**Comment by olamothe** [comment]
Also, @vbernard-coveo was trying to implement this for a particular client with custom code, but got the error that the `CategoryFacet` was executing two query when being reset.

Is this something that works properly when you include a category facet ? Were you able to test with it ?

File: src/ui/Omnibox/Omnibox.ts (hunk 2 above)
Comment (ThibodeauJF) [reply src/ui/Omnibox/Omnibox.ts]: Well technically if you have a Querybox, you don't have an Omnibox, except if I go put that piece of code in the Querybox too or a level higher in the Searchbox 🤔

**Comment by ThibodeauJF** [comment]
The clearing breadcrumb itself doesn't trigger any queries. We just clear all the facets when a new query is triggered from a specific list of components. Tested with category facet and things work well.

**Comment by olamothe** [comment]
Perfect !

File: src/ui/Omnibox/Omnibox.ts (hunk 2 above)
Comment (ThibodeauJF) [reply src/ui/Omnibox/Omnibox.ts]: @olamothe What do you think? Fine as it is? I don't think the Querybox is used that much I believe anymore... Most new clients will have the Searchbox/Omnibox package...

**Pull request merged**